Both sites already tolerate the errnos an OS reports when it cannot fsync a
directory at all. WASIX reports EISDIR, which the surrounding code recognises
for the open() call above but not for the fsync itself.

--- a/src/backend/storage/file/fd.c
+++ b/src/backend/storage/file/fd.c
@@ -3902,7 +3902,8 @@
 	 * Some OSes don't allow us to fsync directories at all, so we can ignore
 	 * those errors. Anything else needs to be logged.
 	 */
-	if (returncode != 0 && !(isdir && (errno == EBADF || errno == EINVAL)))
+	if (returncode != 0 && !(isdir && (errno == EBADF || errno == EINVAL ||
+									   errno == EISDIR)))
 	{
 		int			save_errno;
 
--- a/src/common/file_utils.c
+++ b/src/common/file_utils.c
@@ -435,7 +435,8 @@
 	 * Some OSes don't allow us to fsync directories at all, so we can ignore
 	 * those errors. Anything else needs to be reported.
 	 */
-	if (returncode != 0 && !(isdir && (errno == EBADF || errno == EINVAL)))
+	if (returncode != 0 && !(isdir && (errno == EBADF || errno == EINVAL ||
+									   errno == EISDIR)))
 	{
 		pg_log_error("could not fsync file \"%s\": %m", fname);
 		(void) close(fd);
